src/storage/storage_structure/disk_overflow_file.cpp

```diff
@@ -49,14 +51,19 @@ void DiskOverflowFile::setStringOverflowWithoutLock(
                 BufferPoolConstants::PAGE_4KB_SIZE, strlen(srcRawString)));
     }
     addNewPageIfNecessaryWithoutLock(len);
-    auto updatedPageInfoAndWALPageFrame = createWALVersionOfPageIfNecessaryForElement(
+    auto originalPageCursor = PageUtils::getPageElementCursorForPos(
         nextBytePosToWriteTo, BufferPoolConstants::PAGE_4KB_SIZE);
-    memcpy(updatedPageInfoAndWALPageFrame.frame + updatedPageInfoAndWALPageFrame.posInPage,
-        srcRawString, len);
-    DBFileUtils::unpinWALPageAndReleaseOriginalPageLock(
-        updatedPageInfoAndWALPageFrame, *fileHandle, *bufferManager, *wal);
-    TypeUtils::encodeOverflowPtr(diskDstString.overflowPtr,
-        updatedPageInfoAndWALPageFrame.originalPageIdx, updatedPageInfoAndWALPageFrame.posInPage);
+    bool insertingNewPage = false;
+    if (originalPageCursor.pageIdx >= fileHandle->getNumPages()) {
```

Comment:
Isn't the logic here serving similar functionality with `addNewPageIfNecessaryWithoutLock(len);` ? Why separate them in different places?
Also, I think you also need to check if the `originalPageCursor.elemPosInPage + len` is exceeding 4KB.

Reply:
Yes, indeed. I was trying to keep things as close as possible to the old code since it wasn't working, however we may as well clean up that too.
I think the existing checks are sufficient. It's checking that the length is no greater than 4KB, and if `originalPageCursor.elemPosInPage + len` is greater than 4KB `addNewPageIfNecessaryWithoutLock` will add a new page.